### PR TITLE
use consistent style guide in this project

### PR DESCRIPTION
**Pull request by PatrickHaussmann**
I think it would be really usefull if this project uses a common style guide.

I don't really care which tool we use. So i would propose using prettier for now. [Why Prettier?](https://prettier.io/docs/en/why-prettier.html)

I have not made any changes except styling. The steps to verify this are: [Install Prettier](https://prettier.io/docs/en/install.html)

1. `npm install --save-dev --save-exact prettier`
1. Add the content of `.prettierrc.json` and `.prettierignore`
1. `npx prettier --write .`

Setting tabWidth to 4 is what I think is most consistent with the code styling that has been used so far.
I am completely open to using alternatives to prettier, changing the settings (e.g. tabWidth) or adding something to prettierignore.

### CODE REVIEW

**Comment by marlon360** [comment]
Hi @PatrickHaussmann!
Thank you for this PR. This was definitely needed!

I know I used a tab width of 4, but I think we should change it to 2. What do you think?

And I would like to add commands for `prettier --check .` and `prettier --write .` to the scripts of `package.json`.

It would be also nice to have a GitHub action that checks the styling for every PR.

**Comment by PatrickHaussmann** [comment]
I think tab width of 2 is fine, too.

What do you mean with "add commands to [..] package.json"?

Should I make a new pull request? If not every single file in the project would be changed twice

**Comment by marlon360** [comment]
In package.json we have scripts:
```
 "scripts": {
        "test": "echo \"Error: no test specified\" && exit 1",
       "start": "npm run build && node dist/server.js",
       "build": "tsc && npm run docs:build",
       "docs:dev": "vuepress dev docs",
       "docs:build": "vuepress build docs"
 },
```
I would like to add two more:

```
"lint": "prettier --check .",
"lint:fix": "prettier --write .",
```

> Should I make a new pull request? If not every single file in the project would be changed twice

You can commit it here, that's no problem.

**Comment by PatrickHaussmann** [comment]
I have now done that. are there any comments or suggestions for improvement on this?

File: .github/workflows/code_style.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    build:
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout Repo Files
+          uses: actions/checkout@v2
+        - name: Download and setup npm
+          uses: actions/setup-node@v2
+        - name: Install prettier
+          run: npm install -g prettier
+        - name: Chekc code style with prettier
```
Comment (Rubber1Duck) [inline .github/workflows/code_style.yml]: typo in "check"! (is "Chekc")

**Review by Rubber1Duck — CHANGES_REQUESTED** [review]

**Comment by PatrickHaussmann** [comment]
Installation of prettier fails with error: "Process completed with exit code 243".

Should the command simply be `npm install`?
prettier is already included in package.json

**Comment by Rubber1Duck** [comment]
> Should the command simply be `npm install`?

I think so, globaly installation needs root rights!

**Comment by marlon360** [comment]
Let's use `npm install` and after that `npm run lint`.

**Comment by PatrickHaussmann** [comment]
I am sorry. I saw your comment too lat
I think `npm install` does not install the devDependencies

**Comment by marlon360** [comment]
I cannot access the log. Does anyone know what's the problem now?
Edit: I have it now.

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+# minified svg files:
+src/maps/districts.json
+src/maps/states.json
```
Comment (marlon360) [inline .prettierignore:6]: We also need to add the output directory of the build here.
```
# build folder
dist/
```

**Review by marlon360 — CHANGES_REQUESTED** [review]

**Review by marlon360 — APPROVED** [review]

**Pull request merged**